python/cudf/cudf/_lib/stream_compaction.pyx

```diff
@@ -77,6 +79,29 @@ def drop_duplicates(in_index, in_cols, subset=None, keep='first'):
     return (out_cols[:-1], out_cols[-1])
 
 
+def nunique(in_col):
+    """
+    get number of unique elements in the input column
+
+    in_col: input column to check for unique element count
+
+    out_count: number of unique elements in the input column
+    """
+    in_col_null = in_col
+    # convert nans_to_nulls
+    if (in_col.dtype in [np.float16, np.float32, np.float64]):
+        in_col_null = in_col.set_mask(nans_to_nulls(in_col))
+    check_gdf_compatibility(in_col_null)
+    cdef gdf_column* c_in_col = column_view_from_column(in_col_null)
+    cdef int count = 0
+    with nogil:
+        count = cpp_unique_count(c_in_col[0])
+    # if null and nan both are present, increment count by 1
+    if in_col.null_count>0 and in_col_null.null_count > in_col.null_count:
+        count = count + 1
```

Comment:
Ideally these two lines could be somehow moved into the C++ layer and parameterized for handling `NaN`?

Reply:
Agree

Reply:
working on `NaN` vs null in sorting in https://github.com/rapidsai/cudf/issues/2191. Once the requirements are collected for SQL, Spark, & Pandas, I will update it in C++ code. 
shall I keep this until that is ready?

Reply:
Assuming that will be finished for 0.10 then yes, I would keep this open until you can satisfy all requirements.

Reply:
moved this to c++. (added a TODO comment to fix after NaN support for equality_operator)